### DISCUSSION TITLE
Infer type (query or mutation) using GetInferenceHelpers

### DISCUSSION THREAD

**Discussion by martin-fv** [trpc/trpc#2962] [General]
GetInferenceHelpers gives me the input and output of a route, but is it possible to infer whether it's a query or mutation so I can mock using get or post.

Appreciate it

**Comment by KATT** [comment]
Not currently, but you can probably hack your own version that does that:

Here's the code for it:

https://github.com/trpc/trpc/blob/ac1e2bb78d89f64bfa0dbb9c844f6e0dd57f05e5/packages/server/src/core/types.ts#L49-L60


You can use `inferProcedureParams<X>` in order to get more metadata about it
